fe/parameterize_lambda.py

```diff
@@ -0,0 +1,199 @@
+        self.outfile = outfile
+        cpdb = app.PDBFile(self.pdb_str)
+        PDBFile.writeHeader(cpdb.topology, self.outfile)
+        self.topology = cpdb.topology
+        self.frame_idx = 0
+
+    def write(self, x):
+        if self.outfile is None:
+            raise ValueError("remember to call write_header first")
+        self.frame_idx += 1
+        PDBFile.writeModel(self.topology, x, self.outfile, self.frame_idx)
+
+    def close(self):
+        PDBFile.writeFooter(self.topology, self.outfile)
+        self.outfile.flush()
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Quick Test')
+    parser.add_argument('--out_pdb', type=str)
+
+    # parser.add_argument('--precision', type=str)    
+    parser.add_argument('--complex_pdb', type=str)
+    parser.add_argument('--solvent_pdb', type=str)
+    parser.add_argument('--ligand_sdf', type=str)
+    args = parser.parse_args()
+
+    amber_ff = app.ForceField('amber99sb.xml', 'tip3p.xml')
+
+    # host_pdb = app.PDBFile(args.complex_pdb)
+    # modeller = app.Modeller(host_pdb.topology, host_pdb.positions)
+    # modeller.addSolvent(amber_ff, numAdded=6000)
+    # PDBFile.writeFile(modeller.topology, modeller.positions, open("sanitized.pdb", 'w'))
+    # assert 0
+
+    suppl = Chem.SDMolSupplier(args.ligand_sdf, removeHs=False)
+    for guest_mol in suppl:
+        break
+
+    T = 10000
+    dt = 0.0015
+    step_sizes = np.ones(T)*dt
+    cas = np.ones(T)*0.99
+
+    num_gpus = 1
+    pool = multiprocessing.Pool(num_gpus)
+
+    all_du_dls = []
+
+    lambda_schedule = np.linspace(0.00001, 0.99999, num=T)
+
+    epoch = 0
+
+    init_conf = guest_mol.GetConformer(0)
+    init_conf = np.array(init_conf.GetPositions(), dtype=np.float64)
+    init_conf = init_conf/10 # convert to md_units
+    conf_com = com(init_conf)
+    init_mol = Chem.Mol(guest_mol)
+
+    for host_idx, host_pdb_file in enumerate([args.complex_pdb, args.solvent_pdb]):
+
+        host_pdb = app.PDBFile(host_pdb_file)
+
+        if host_idx == 0:
+            host_name = "complex"
+        elif host_idx == 1:
+            host_name = "solvent"
+
+        for mode in ['insertion', 'deletion']:
+            print("Mode", mode)
+
+            host_conf = []
+            for x,y,z in host_pdb.positions:
+                host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
+
+            host_conf = np.array(host_conf)
+            init_combined_conf = np.concatenate([host_conf, init_conf])
+
+            perm = hilbert_sort(init_combined_conf)
+            sim = simulation.Simulation(
+                guest_mol,
+                host_pdb,
+                mode,
+                step_sizes,
+                cas,
+                lambda_schedule,
+                perm
+            )
+
+            num_conformers = 1
+
+            guest_mol.RemoveAllConformers()
+            AllChem.EmbedMultipleConfs(guest_mol, num_conformers, randomSeed=2020)
```

Comment:
You should check for failure here, I think if the result is -1 no confs were generated.